File: C#/Remove Element/RemoveElement.cs
public class Solution {
    public int RemoveElement (int[] nums, int val) {
        if (nums.Length == 0)
            return 0;
        int ptr = 0;
        for (int j = 0; j < nums.Length; j++) {
            if (ptr != j) {
                nums[ptr] = nums[j];
            }
            if (nums[j] != val) {
                ptr++;
            }
        }
        return ptr;
    }
}

File: Python/Remove Element/RemoveElement.py
class Solution:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        if (len(nums) == 0):
            return 0
        ptr = 0
        for i in range(len(nums)):
            if (nums[i] != val):
                nums[ptr] = nums[i]
                ptr += 1
        return ptr